### ISSUE TITLE
SA1515 falsely (?) raised for comments following an attribute

### ISSUE THREAD

**Issue by jackwhelpton** [DotNetAnalyzers/StyleCopAnalyzers#1490]
I'm seeing a violation for the following code (which previously passed the StyleCop "classic" version of this rule):

````csharp
[ContractClassFor(typeof(ILinkActivator<>))]
// ReSharper disable once InconsistentNaming
// Contract class containing only metadata pertaining to an interface.
// Using naming convention adopted by the Code Contracts team for .NET framework contracts.
public abstract class ILinkActivatorContract<T> : ILinkActivator<T> where T : LinkTemplate
````

The comment pertains to the class and not the attribute, and because of how ReSharper works needs to be adjacent to the code element it addresses: in this case it is suppressing a warning about the naming of the class.

An alternative that now passes would be:

````csharp
[ContractClassFor(typeof(ILinkActivator<>))]

// ReSharper disable once InconsistentNaming
// Contract class containing only metadata pertaining to an interface.
// Using naming convention adopted by the Code Contracts team for .NET framework contracts.
public abstract class ILinkActivatorContract<T> : ILinkActivator<T> where T : LinkTemplate
````

but I'm not keen on having the attribute visually disjoint from the class to which it applies.

**Comment by jackwhelpton** [comment]
Another, less justifiable example that used to pass SA1515:

````csharp
Contract.Requires(context != null);
// ReSharper disable once CompareNonConstrainedGenericWithNull
// Intentionally only guarding against null arguments; for value types, default(T) is acceptable.
Contract.Requires(parameters != null);
````

The argument for omitting the space was so that the contract block was clearly one unit and visually distinct from the rest of the method, but in hindsight I suspect it's more readable with the whitespace present.

**Comment by sharwell** [comment]
What about using a block comment?

````csharp
[ContractClassFor(typeof(ILinkActivator<>))]
/* ReSharper disable once InconsistentNaming
 * Contract class containing only metadata pertaining to an interface.
 * Using naming convention adopted by the Code Contracts team for .NET framework contracts.
 */
public abstract class ILinkActivatorContract<T> : ILinkActivator<T> where T : LinkTemplate
````

**Comment by jackwhelpton** [comment]
That's a good compromise, cheers. I'd lean towards putting ```*/``` at the end of the last line; as is the attribute still looks a little like it's floating above the class. This makes me wonder about a related point: are there style rules governing multiline comments?

**Comment by sharwell** [comment]
> This makes me wonder about a related point: are there style rules governing multiline comments?

Not right now, that I know of.
